app/src/main/java/com/nowak/wjw/simplecompass/ui/main/MainViewModel.java

```diff
@@ -35,19 +40,93 @@
     public LiveData<Integer> buttonTextResId;
     public LiveData<Boolean> editTextVisible;
     public LiveData<Boolean> hideKeyBoard;
+    public LiveData<Boolean> showLocationError;
+    private MutableLiveData<Boolean> mShouldProvideRationale = new MutableLiveData<>(false);
+    public LiveData<Event<Boolean>> shouldProvideRationale = Transformations.map(mShouldProvideRationale, should -> new Event(should));
```

Comment:
lambda ?

Reply:
pytasz, czy sugerujesz?
jak dla mnie to lambda jest ;P

Reply:
można to uproscic

Reply:
nie jest to dla mnie widoczne na pierwszy rzut oka, zastanowie sie i dam znać

Reply:
chodzi mi o to że można zrobić odwołanie do elementu klasy, u ciebie by to było Event::new